bazel_tools/proto.bzl

```diff
@@ -159,3 +165,95 @@ proto_gen = rule(
 
 def _is_windows(ctx):
     return ctx.configuration.host_path_separator == ";"
+
+def maven_tags(group, artifact_prefix, artifact_suffix):
+    if group and artifact_prefix and artifact_suffix:
+        return ["maven_coordinates=%s:%s-%s:__VERSION__" % (group, artifact_prefix, artifact_suffix)]
+    else:
+        return []
+
+def proto_jars(
+        name,
+        srcs,
+        strip_import_prefix = "",
+        deps = [],
+        proto_deps = [],
+        java_deps = [],
+        scala_deps = [],
+        file_root = None,
+        javadoc_root_packages = [],
+        maven_group = None,
+        maven_artifact_prefix = None,
+        maven_java_artifact_suffix = "java-proto"):
+    # Tarball containing the *.proto files.
+    pkg_tar(
+        name = "%s_src" % name,
+        srcs = srcs,
+        extension = "tar.gz",
+        strip_prefix = strip_import_prefix,
+        package_dir = file_root,
+        visibility = ["//visibility:public"],
```

Comment:
Why default to public visibility everywhere? This widens the visibility of some targets seemingly unnecessarily.

Reply:
For convenience. These things are often published by `//release` and I was lazy. I tried copying the existing visibility but it meant having 2 or 3 `visibility` parameters, which got unwieldy.

Do you have a better approach in mind?

Reply:
I'd set a single `visibility` parameter to `proto_jars` and forward it to the rules that are actually dependencies of other targets. The rest I'd just leave at the default visibility, which is private unless specified otherwise on the package level.

Grepping through the code base it looks like the following rules could remain private: `pkg_tar`, `javadoc_library`, `proto_get`, `scala_source_jar`, `scaladoc_jar`. They are either only used within `proto_jars`, or by the release script which calls `bazel build` directly and therefore isn't affected by target visibility.

Reply:
https://github.com/digital-asset/daml/pull/8121 should take care of this.